packages/blade/src/components/Carousel/CarouselContext.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import type { CarouselProps } from './types';
+
+type CarouselContextProps =
+  | (Pick<
+      CarouselProps,
+      'visibleItems' | 'carouselItemWidth' | 'shouldAddStartEndSpacing' | 'carouselItemAlignment'
+    > & {
+      carouselContainerRef: React.RefObject<HTMLDivElement>;
+      setActiveIndicator: (value: React.SetStateAction<number>) => void;
+      carouselId: string | null;
+      totalNumberOfSlides: number;
+      activeSlide: number;
+    })
+  | null;
+const CarouselContext = React.createContext<CarouselContextProps>(null);
+
+const useCarouselContext = (): NonNullable<CarouselContextProps> => {
+  const state = React.useContext<CarouselContextProps>(CarouselContext);
+
+  if (!state) {
+    throw new Error('[Blade Carousel]: useCarouselContext must be used within Carousel');
```

Comment:
well thats a bummer. 

<img width="447" alt="image" src="https://github.com/razorpay/blade/assets/35374649/2be10336-8484-4627-bfc8-ed68a7888ca2">

Reply:
Used `throwBladeError`, For now did `return state!`. 

This happens because TypeScript doesn't know that throwBladeError will stop execution, thus TS doesn't narrow the type. we would probably need to make it into an [`assertion` function](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions).